src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java

```diff
@@ -103,33 +99,44 @@ OffenderMatchDetail getOffenderMatchDetail(String crn) {
         log.debug("Looking for offender detail for CRN :{}", crn);
         return Mono.zip(offenderRestClient.getOffenderMatchDetailByCrn(crn),
                         offenderRestClient.getConvictionsByCrn(crn)
-                            .onErrorResume(OffenderNotFoundException.class, e -> Mono.just(Collections.emptyList())),
+                                .onErrorResume(OffenderNotFoundException.class, e -> Mono.just(Collections.emptyList())),
                         offenderRestClient.getProbationStatusByCrn(crn))
-            .map(tuple -> addMostRecentEventToOffenderMatch(tuple.getT1(), tuple.getT2(), tuple.getT3()))
-            .block();
+                .map(tuple -> addMostRecentEventToOffenderMatch(tuple.getT1(), tuple.getT2(), tuple.getT3()))
+                .block();
     }
 
     private OffenderMatchDetail addMostRecentEventToOffenderMatch(OffenderMatchDetail offenderMatchDetail,
-                                                          List<Conviction> convictions,
-                                                          ProbationStatusDetail probationStatus) {
+                                                                  List<Conviction> convictions,
+                                                                  ProbationStatusDetail probationStatus) {
 
         if (offenderMatchDetail == null) {
             return null;
         }
 
         Sentence sentence = Optional.ofNullable(convictions).orElse(Collections.emptyList()).stream()
-            .filter(Conviction::getActive)
-            .findFirst()
-            .map(Conviction::getSentence)
-            .orElse(getSentenceForMostRecentConviction(convictions));
+                .filter(Conviction::getActive)
+                .findFirst()
+                .map(Conviction::getSentence)
+                .orElse(getSentenceForMostRecentConviction(convictions));
 
         return OffenderMapper.offenderMatchDetailFrom(offenderMatchDetail, sentence, probationStatus);
     }
 
     Sentence getSentenceForMostRecentConviction(List<Conviction> convictions) {
         return Optional.ofNullable(convictions).orElse(Collections.emptyList()).stream()
-            .min(Comparator.comparing(Conviction::getConvictionDate, Comparator.nullsLast(Comparator.reverseOrder())))
-            .map(Conviction::getSentence)
-            .orElse(null);
+                .min(Comparator.comparing(Conviction::getConvictionDate, Comparator.nullsLast(Comparator.reverseOrder())))
+                .map(Conviction::getSentence)
+                .orElse(null);
+    }
+
+    public Mono<GroupedOffenderMatchesEntity> getGroupedOffenderMatchesByDefendantIdAndGroupId(String defendantId, Long groupId) {
+        log.debug("Entered getGroupedMatchesByDefendantId with defendantId {} , groupId {}", defendantId, groupId);
```

Comment:
Not find any log statement in the codebase so added one. if it's ok, I could add in other classes. please let me know.

Reply:
A risk with this kind of logging is that the logging doesn't actually reflect reality - the method name is wrong 😬 
```suggestion
        log.debug("Entered getGroupedOffenderMatchesByDefendantIdAndGroupId with defendantId {} , groupId {}", defendantId, groupId);
```

If it's a pattern we choose to adopt we could probably use some kind of annotation that works out the specifics for you 🤔

Reply:
AppInsights is there already so removed it.